### DISCUSSION TITLE
qscintilla2 is in linuxbrew but only compiles on macOS

### DISCUSSION THREAD

**Discussion by lteneyck** [Homebrew/discussions#317] [Linux]
The qscintilla2 formula contains a "spec" parameter for customizing the build for different operating systems.

The present formula under def install, only tests if the compiler is clang or not, and sets spec to either "macx-clang" or "macx-g++".

This breaks the configure step on ubuntu 20.04, and I expect on any linux system. I edited the formula to force spec="linux-g++" the the formula builds on my system, but of course will now be broken for all Mac systems. Clearly there are flags that are set to tell the install scripts what kind of system they are on, but I don't know what they are and haven't found any documentation for them. It does seem that with this information a one-line patch should clear up the situation.

I need this to build octave, because the ubuntu version of octave is a snap, and the sandbox limits on snaps prevent me from doing some necessary things for my application. 

Thanks in advance.

**Comment by carlocab** [comment]
Thanks for pointing this out. Could you open a PR to fix this? If you want to run a command only on macOS,
```
on_macos do
  command
end
```
and on Linux,
```
on_linux do
  command
end
```

**Comment by lteneyck** [comment]
Thank you for the information. Right now I am rebuilding my systems but can
get to this tomorrow. I am a git noobie but have been around systems for a
long time. If you can point me to a quick description of pull requests it
would be helpful and then may be useful for other things I run across.

**Reply by carlocab** [reply]
You may find the following helpful.

https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
https://docs.brew.sh/Formula-Cookbook

Best to start with the first link and keep the second as a reference. If you're still not sure how to do something, check out other formulae to see how they do it.

**Comment by lteneyck** [comment]
Excellent, thank you very much!

**Comment by danielnachun** [comment]
I already have a PR open to fix this, so that octave can build on Linux: https://github.com/Homebrew/linuxbrew-core/pull/21834.  There's currently an unrelated issue holding that up, but when it is approved it should work.  You can look at the changes I made and modify the formula with `brew edit qscintilla2` if you need `qscintilla2` urgently.

**Comment by lteneyck** [comment]
Thank you! I will look at both of your pull requests, because the only
reason I need qscintilla2 is so I can get Octave.

With my qscintilla2 patch on Ubuntu 20.04 (which was to just stick
spec="linux-g++" into the formula, hence my request for instruction) I was
able to get octave to compile, but when I ran the "sombrero" test case, I
got the graphics window with no text -- which is probably due to the fact
that qt no longer ships fonts. I have made no further tests of that octave
because I need full graphics.

Lynn Ten Eyck

On Sun, Dec 20, 2020 at 11:11 PM Daniel Nachun <notifications@github.com>
wrote:

> I already have a PR open to fix this, so that octave can build on Linux:
> Homebrew/linuxbrew-core#21834
> <https://github.com/Homebrew/linuxbrew-core/pull/21834>. There's
> currently an unrelated issue holding that up, but when it is approved it
> should work. You can look at the changes I made and modify the formula with brew
> edit qscintilla2 if you need qscintilla2 urgently.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Homebrew/discussions/discussions/317#discussioncomment-228535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2CQZLKWDI7IZ244WJNF53SV3YL3ANCNFSM4VBZMSUQ>
> .
>


-- 
Lynn F. Ten Eyck
Lynn.TenEyck@gmail.com

**Reply by danielnachun** [reply]
As you'll see the Octave formula, there were also issues with OpenGL support that had to be fixed, which might have explained your issues with the graphics windows.  Let me know if there are still issues after trying the changes I made!

**Comment by lteneyck** [comment]
Which graphics terminal were you using with octave? I have been using qt on
other systems.

Lynn

On Mon, Dec 21, 2020 at 11:57 AM Daniel Nachun <notifications@github.com>
wrote:

> As you'll see the Octave formula, there were also issues with OpenGL
> support that had to be fixed, which might have explained your issues with
> the graphics windows. Let me know if there are still issues after trying
> the changes I made!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Homebrew/discussions/discussions/317#discussioncomment-230836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2CQZLRJOSJ73VZVHIIE5TSV6SCRANCNFSM4VBZMSUQ>
> .
>


-- 
Lynn F. Ten Eyck
Lynn.TenEyck@gmail.com

**Reply by danielnachun** [reply]
I have only been using it in a headless server environment, and only to run a MATLAB program I'm trying to port to a different language.  I only noticed the OpenGL issues because the build was failing, and it's possible that the graphical issues you are having aren't related.  You may be correct that missing fonts are a problem here.  You could try to install them through the homebrew-fonts tap: https://github.com/Linuxbrew/homebrew-fonts.
